pkg/controller/account/iam.go

```diff
@@ -0,0 +1,185 @@
+		"sessionId":    *GetFederationTokenOutput.Credentials.AccessKeyId,
+		"sessionKey":   *GetFederationTokenOutput.Credentials.SecretAccessKey,
+		"sessionToken": *GetFederationTokenOutput.Credentials.SessionToken,
+	}
+
+	jsonCredentials, err := json.Marshal(federationCredentials)
+
+	// TODO better error here
+	if err != nil {
+		fmt.Println("Error serializing json")
+		fmt.Println(err)
+		return "", err
+	}
+
+	// Build URL to request Signin Token via Federation end point
+	baseFederationURL, err := url.Parse(federationEndPointURL)
+
+	if err != nil {
+		fmt.Println("Malformed URL: ", err.Error())
+		return "", err
+	}
+
+	federationParams := url.Values{}
+
+	federationParams.Add("Action", "getSigninToken")
+	federationParams.Add("SessionType", "json")
+	federationParams.Add("Session", string(jsonCredentials))
+
+	baseFederationURL.RawQuery = federationParams.Encode()
+
+	// Make HTTP request to retrieve Federated Signin Token
+	res, err := http.Get(baseFederationURL.String())
+
+	if err != nil {
+		fmt.Printf("Error requesting Signin Token from: %s\n", baseFederationURL.String())
+		return "", err
+	}
+
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		fmt.Printf("Unable to read response body: %s", baseFederationURL.String())
+		return "", err
+	}
+
+	var SigninResponse awsSigninTokenRepsonse
+
+	// Unmarshal JSON response so we can extract the signin token
+	err = json.Unmarshal(body, &SigninResponse)
+
+	if err != nil {
+		fmt.Printf("Error unmarshalling Federated Signin Response JSON")
+		return "", err
+	}
+
+	signinFederationURL, err := url.Parse(federationEndPointURL)
+	if err != nil {
+		fmt.Println("Malformed URL: ", err.Error())
+	}
+
+	signinParams := url.Values{}
+
+	signinParams.Add("Action", "login")
+	signinParams.Add("Destination", awsConsoleURL)
+	signinParams.Add("Issuer", issuer)
+	signinParams.Add("SigninToken", SigninResponse.SigninToken)
+
+	signinFederationURL.RawQuery = signinParams.Encode()
+
+	// Return Signin Token
+	return signinFederationURL.String(), nil
+
+}
+
+// BuildSTSUser takes all parameters required to create a user, user secret
+func (r *ReconcileAccount) BuildSTSUser(reqLogger logr.Logger, awsSetupClient awsclient.Client, awsClient awsclient.Client, account *awsv1alpha1.Account, accountID string, nameSpace string) (string, error) {
+	reqLogger.Info("Creating IAM STS User")
+
+	// If IAM user was just created we cannot immediately create STS credentials due to an issue
+	// with eventual consisency on AWS' side
+	time.Sleep(10 * time.Second)
+
+	// Create STS user for SRE admins
+	STSCredentials, STSCredentialsErr := getStsCredentials(reqLogger, awsClient, "", accountID)
+	if STSCredentialsErr != nil {
+		reqLogger.Info("Failed to get SRE admin STSCredentials from AWS api ", "Error", STSCredentialsErr.Error())
+		return "", STSCredentialsErr
+	}
+
+	STSUserName := account.Name + "-STS"
+
+	IAMAdministratorPolicy := "arn:aws:iam::aws:policy/AdministratorAccess"
```

Comment:
Should this be a constant ?

Reply:
It can't be because `sts.PolicyDescriptorType` takes a pointer :(